{
  Throwable exception=new Throwable("failed");
  ListenableFuture<String> future=immediateFailedFuture(exception);
  ListenableFuture<List<String>> compound=allAsList(ImmutableList.of(future));
  try {
    getDone(compound);
    fail();
  }
 catch (  ExecutionException expected) {
    assertSame(exception,expected.getCause());
  }
}
